src/app/clusters/boolean-state-configuration-server/boolean-state-configuration-server.cpp

```diff
@@ -0,0 +1,443 @@
+Delegate * gDelegateTable[kBooleanStateConfigurationDelegateTableSize] = { nullptr };
+
+Delegate * GetDelegate(EndpointId endpoint)
+{
+    uint16_t ep = emberAfGetClusterServerEndpointIndex(endpoint, BooleanStateConfiguration::Id,
+                                                       EMBER_AF_BOOLEAN_STATE_CONFIGURATION_CLUSTER_SERVER_ENDPOINT_COUNT);
+    return (ep >= kBooleanStateConfigurationDelegateTableSize ? nullptr : gDelegateTable[ep]);
+}
+
+bool isDelegateNull(Delegate * delegate)
+{
+    if (delegate == nullptr)
+    {
+        return true;
+    }
+    return false;
+}
+
+class BooleanStateConfigAttrAccess : public AttributeAccessInterface
+{
+public:
+    BooleanStateConfigAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), BooleanStateConfiguration::Id) {}
+
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR WriteCurrentSensitivityLevel(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder);
+    CHIP_ERROR ReadCurrentSensitivityLevel(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder);
+};
+
+BooleanStateConfigAttrAccess gAttrAccess;
+
+CHIP_ERROR BooleanStateConfigAttrAccess::WriteCurrentSensitivityLevel(const ConcreteDataAttributePath & aPath,
+                                                                      AttributeValueDecoder & aDecoder)
+{
+    uint8_t curSenLevel;
+    ReturnErrorOnFailure(aDecoder.Decode(curSenLevel));
+    // VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == SupportedSensitivityLevels::Get(aPath.mEndpointId, &supportedSensLevel),
+    //                     CHIP_IM_GLOBAL_STATUS(Failure));
+
+    // VerifyOrReturnError(curSenLevel < supportedSensLevel, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    // ReturnErrorOnFailure(GetSafeAttributePersistenceProvider()->WriteScalarValue(
+    //     ConcreteAttributePath(aPath.mEndpointId, aPath.mClusterId, CurrentSensitivityLevel::Id), curSenLevel));
+
+    return StoreCurrentSensitivityLevel(aPath.mEndpointId, curSenLevel);
+}
+
+CHIP_ERROR BooleanStateConfigAttrAccess::ReadCurrentSensitivityLevel(const ConcreteReadAttributePath & aPath,
+                                                                     AttributeValueEncoder & aEncoder)
+{
+    uint8_t senLevel;
+    CHIP_ERROR err = GetSafeAttributePersistenceProvider()->ReadScalarValue(
+        ConcreteAttributePath(aPath.mEndpointId, aPath.mClusterId, CurrentSensitivityLevel::Id), senLevel);
+    if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        uint8_t supportedSensLevel;
+        VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == SupportedSensitivityLevels::Get(aPath.mEndpointId, &supportedSensLevel),
+                            CHIP_IM_GLOBAL_STATUS(Failure));
+        senLevel = static_cast<uint8_t>(supportedSensLevel - 1);
```

Comment:
`supportedSensLevel` can never be 0 here?

Reply:
The spec mandates it to be in the range 2 to 10, but a misconfigured device could set it to 0 so for the sake of being on the safe side we could make a verification that supportedSensLevel is not < 2, before we do the operation? 🤔

Reply:
Since SupportedSensLevel has an F quality - I guess it can only be validated during certification? But I could validate the math operation here not to cause other side effects.